### PR TITLE
CLI enhancements: Add --hooks, --filename options

### PR DESCRIPTION
**Pull request by msutkowski**
Changes from the base PR at #8:

- Supports URLs or local files in the CLI (in the current version, you cannot generate from a URL even though SwaggerParser supports it)
- Accepts a file output arg as an alternative to using the `file > outputFile.ts` method
- Runs prettier on the generated code, and will follow a user's prettier config when run in a dir with a prettier config
- Adds prettier/husky to the repo to standardize code formatting
- Moves lodash `upperFirst` to `capitalize` from rtk-query so we have the same behavior

### CODE REVIEW

**Comment by msutkowski** [comment]
I wanted to tack on to @kahirokunn's PR with a small amount of cleanup and some CLI improvements. Please let me know what you think.

**Comment by msutkowski** [comment]
@kahirokunn Please take a peek at this and let me know what you think. I hope this doesn't bother you, but it was just easier to do this instead of leaving a bunch of PR comments. Thanks again for your work on #8

File: src/bin/cli.ts [hunk 1]
```diff
@@ -18,12 +20,15 @@ program
...
+  .parse(process.argv);
 
 if (program.args.length === 0) {
-  program.help()
+  program.help();
 } else {
-  const schemaAbsPath = path.resolve(process.cwd(), program.args[0])
+  const schemaLocation = program.args[0];
+
+  const schemaAbsPath = isValidUrl(schemaLocation) ? schemaLocation : path.resolve(process.cwd(), schemaLocation);
```
Comment (kahirokunn) [inline src/bin/cli.ts:31]: This is a great idea! ❤️

File: src/bin/cli.ts [hunk 2]
```diff
@@ -18,12 +20,15 @@ program
   .option('--responseSuffix <name>', 'pass response suffix')
   .option('--baseUrl <url>', 'pass baseUrl')
   .option('-h, --hooks', 'generate React Hooks')
-  .parse(process.argv)
+  .option('--file <filename>', 'output file name (ex: generated.api.ts)')
```
Comment (kahirokunn) [inline src/bin/cli.ts:23]: a lot of people want this option!

File: src/generators/react-hooks.ts [hunk 3]
```diff
@@ -1,41 +1,42 @@
...
-import * as ts from "typescript";
-import { factory } from "typescript";
+import * as ts from 'typescript';
+import { getOperationName } from 'oazapfts/lib/codegen/generate';
+import { capitalize, isQuery } from '../utils';
+import { OperationDefinition } from '../types';
 
-import { OperationDefinition } from "../type";
-import { isQuery } from "../utils/isQuery";
+const { factory } = ts;
```
Comment (kahirokunn) [inline src/generators/react-hooks.ts:6]: nice 👀

**Review by kahirokunn — APPROVED** [review]
@msutkowski
This is awesome!
LGTM!

**Pull request merged**